src/server/ua_subscription_events.c

```diff
@@ -519,6 +521,60 @@ static const UA_NodeId isInFolderReferences[2] =
     {{0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_ORGANIZES}},
      {0, UA_NODEIDTYPE_NUMERIC, {UA_NS0ID_HASCOMPONENT}}};
 
+#ifdef UA_ENABLE_PUBSUB_EVENTS
+static UA_StatusCode
+insertDataValueIntoDSWQueue(UA_Server *server, UA_DataSetWriter *dsw, UA_DataValue *value)  {
+    if(dsw->eventQueueEntries == dsw->eventQueueMaxSize){
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                     "EventQueueMaxSize reached %lu / %lu", dsw->eventQueueEntries, dsw->eventQueueMaxSize);
```

Comment:
I think a warning is enough. Just drop the new field or remove the oldest entry and add this new one.